packages/flutter/test/material/theme_test.dart

```diff
@@ -265,4 +265,40 @@ void main() {
     expect(glyphText.text.style.color, Colors.orange);
     expect(glyphText.text.style.fontSize, 20.0);
   });
+
+  testWidgets(
+    'Same ThemeData reapplied does not trigger didChangeDepedencies',
+    (WidgetTester tester) async {
+      testBuildCalled = 0;
+      final ThemeData themeData = new ThemeData(primaryColor: const Color(0xFF000000));
+
+      await tester.pumpWidget(
+        new MaterialApp(
+          theme: themeData,
+          home: new Test(),
+        ),
+      );
+      expect(testBuildCalled, 1);
+      tester.element(find.byType(MaterialApp)).markNeedsBuild();
```

Comment:
I would not have expected this to trigger the bug even without your fix. The child doesn't change, so the system should short-circuit it and not rebuild it, and the child doesn't call Theme.of(), so it shouldn't get artificially rebuilt when the Theme is rebuilt.

I think a better test would be to just have an explicit Theme widget (no MaterialApp) with a child that is a `const Test()` whose build calls `Theme.of`, which you pumpWidget twice (with the same exact tree and the same instance of themeData). The Test should only be built once. Then you pumpWidget again but with a new instance of ThemeData with the same values. Test should still not be built again. Then you pumpWidget a fourth time, same tree, but with a slightly different instance of ThemeData. Now finally Test should rebuild.